model-optimizer/mo/middle/passes/eliminate.py

```diff
@@ -90,6 +90,9 @@ def mark_undead_nodes(graph, undead_types: list):
     # mark specifically defined with node type set of nodes
     for type in undead_types:
         node_of_specific_type = graph.get_nodes_with_attributes(type=type)
+        #  This condition is necessary for internal operations with type=None
```

Comment:
Please, refactor this method to do just one iteration over the graph (instead of 2 + number_of_undead_types). Like this:
```
undead_types_with_result = undead_types + ['Result']
undead_nodes = []
for node in graph.get_op_nodes():
    node_type = node.soft_get('type', node.op)
    if node_type in undead_types_with_result:
         undead_nodes.append(node)

nx.set_node_attributes(G=graph, name='is_undead', values={n: True for n in undead_nodes})
```
So the line 98 will not be needed also.